pkg/labels/labels.go

```diff
@@ -227,11 +235,21 @@ func NewLabel(key string, value string, source string) Label {
 		value = ""
 	}
 
-	return Label{
+	l := Label{
 		Key:    key,
 		Value:  value,
 		Source: source,
 	}
+	if l.Source == LabelSourceCIDR {
+		c, err := LabelToPrefix(l.Key)
+		if err != nil {
+			logrus.WithField("key", l.Key).WithError(err).Error("Failed to parse CIDR label: invalid prefix.")
```

Comment:
I understand the intent to log errors here if one occurs, but if we think about where we are in the stack, does it really make sense to log in `pkg/labels/labels.go`? My instinct tells me that if we have a CIDR label then it's not going to be malformed by this point and if it is, we should investigate higher up in the stack where that can happen and prevent it from "leaking" down here. 

The problem with logging an error here is that it's not clear what the user should even do. It's a very low-level detail error log.

Reply:
Really good perspective. I'm not entirely sure what the right course of action is here.

CIDR labels really cannot be malformed; everything that parses them already has proper validation somewhere higher up the stack. I added this error message here just for insurance that we will fail CI if we somehow enter this case.

One more insane option would be to print a stacktrace here (but not panic) if we reach this point. At least in debug mode.

Reply:
This seems like something that should be sanitized much much earlier, then the action item for the user is "Report this to the developers". Ideally no-one ever sees the error, unless we somehow missed some sanitization somewhere.

Better still is to avoid passing strings around internally and instead pass around the type after it was sanitized, but I don't know if that's particularly viable. Maybe if `Label` was an interface and we had typed labels that are either string or `netip.Prefix`...? Getting a bit speculative here.

Reply:
We're stuck parsing strings, since the API *has* to be `LabelArray.Has("cidr.10.0.0.0/8")` as required by the k8s label selector package. So, this warning - which I never expect to happen - has to be here.

Reply:
If we can assume that by the time we reach this code it's impossible to fail to parse and the risk for panic is 0, then let's turn it into the `netip` call to `MustParsePrefix` which doesn't return an error and therefore, remove the log.

Reply:
MustParsePrefix panics on invalid input; I really don’t want that to happen on the off chance someone manages to slip a bad policy in.

No great answer here :-/